### ISSUE TITLE
Relevés non synchronisés

### ISSUE THREAD

**Issue by TheoLechemia** [PnX-SI/gn_mobile_occtax#179]
### Version de l'application

Version d'Occtax-mobile affectée par le bug : 2.2.0
Version de GeoNature utilisée : 2.9.2

### Terminal et Version Android

Marque et modèle du terminal : Samsung A22
Version d'Android : 11

### Description du bug et comportement attendu

Les relevés ne se synchronise pas. La synchronisation descendante (récupération des nomenclature, taxonomie etc...) fonctionne, mais pas celle montante. Au bout de 3 ou 4 synchro, cela se met à fonctionner et le relevés sont envoyés.
Nous avons désactivé la synchronisation périodique.

Voici les logs d'une synchro le 9 puis le 10 juillet. Le 10 juillet on voit qu'il y a deux relevés en plus, qui n'ont pas été synchronisé par la synchro du 9.
On voit un `13:32:10.548 INFO: [fr.geonature.datasync.sync.DataSyncViewModel] no periodic synchronization configured: abort`
[occtax_20220709_000943.log](https://github.com/PnX-SI/gn_mobile_occtax/files/9130549/occtax_20220709_000943.log)
[occtax_20220710_000956.log](https://github.com/PnX-SI/gn_mobile_occtax/files/9130552/occtax_20220710_000956.log)

est-ce que ce sera ça qui bloquerait la synchro ?

**Comment by sgrimault** [comment]
Bonjour @TheoLechemia,

D'après les logs, on dirait qu'aucune synchronisation des données et des relevés n'a été fait le 9 et le 10.
La synchronisation périodique a été désactivée (`13:32:10.548 INFO: [fr.geonature.datasync.sync.DataSyncViewModel] no periodic synchronization configured: abort`), donc ça implique qu'il faut lancer la synchronisation des données et des relevés manuellement.
Autre point, au démarrage l'application n'arrive pas à joindre le serveur pour mettre à jour sa configuration :

```
21:56:50.658 INFO: [fr.geonature.datasync.features.settings.usecase.UpdateSettingsFromRemoteUseCase] updating app configuration from 'https://geonature.ecrins-parcnational.fr'...
21:56:50.750 INFO: [fr.geonature.datasync.api.GeoNatureAPIClientImpl] --> GET https://geonature.ecrins-parcnational.fr/api/gn_commons/t_mobile_apps
21:56:50.768 INFO: [fr.geonature.datasync.api.GeoNatureAPIClientImpl] <-- HTTP FAILED: java.net.UnknownHostException: Unable to resolve host "geonature.ecrins-parcnational.fr": No address associated with hostname
```
Ce n'est pas gênant en soit, mais il ne se passera rien si on tente une synchronisation.

**Comment by DonovanMaillard** [comment]
> ```
> 21:56:50.768 INFO: [fr.geonature.datasync.api.GeoNatureAPIClientImpl] <-- HTTP FAILED: java.net.UnknownHostException: Unable to resolve host "geonature.ecrins-parcnational.fr": No address associated with hostname
> ```

Peut-être un soucis ponctuel de DNS pour ce log là, j'ai eu cette erreur sur des services hier qui ont été résolus rapidement... pas certain que ca vienne du mobile.

**Comment by TheoLechemia** [comment]
Effectivement elle est bizarre cette erreur.. le nom de domaine est bon et l'URL appelé répond..

**Comment by gildeluermoz** [comment]
> 21:56:50.768 INFO: [fr.geonature.datasync.api.GeoNatureAPIClientImpl] <-- HTTP FAILED: java.net.UnknownHostException: Unable to resolve host "geonature.ecrins-parcnational.fr": No address associated with hostname

C'est pas le HTTP au lieu de HTTPS le soucis ?
On dirait que l'appel se fait en HTTP et pas en HTTPS.

**Comment by DonovanMaillard** [comment]
c'est vrai que dans nos tests de la 2.9 avec Camille, j'avais du passer mon instance test en HTTPS pour que ca fonctionne.

**Comment by TheoLechemia** [comment]
Pourquoi l'appel se ferait en HTTP ? 
les logs disent : 
`21:56:50.658 INFO: [fr.geonature.datasync.features.settings.usecase.UpdateSettingsFromRemoteUseCase] updating app configuration from 'https://geonature.ecrins-parcnational.fr'...`
et notre conf est en HTTPS aussi

**Comment by TheoLechemia** [comment]
J'ai réussi à reproduire le bug, voici les logs : 
[occtax_20220719_143003.log](https://github.com/PnX-SI/gn_mobile_occtax/files/9141274/occtax_20220719_143003.log)
On dirait que la route du POST `/api/occtax/releve` n'est pas appelée

**Comment by DonovanMaillard** [comment]
@sgrimault j'ai vu que la 2.4 comprend une validation de l'authentification dans le changelog. Est-ce que ca permet notamment de gérer ce cas de figure ? Si oui on peut fermer le ticket.

**Comment by sgrimault** [comment]
@DonovanMaillard,
Non, la correction en question porte sur https://github.com/PnX-SI/gn_mobile_occtax/issues/184 afin de vérifier si l'utilisateur connecté est bien valide coté "Occtax".
